exec/bigmachine.go

```diff
@@ -632,6 +635,9 @@ func (w *worker) Run(ctx context.Context, req taskRunRequest, reply *taskRunRepl
 			if task != nil {
 				task.Error(errors.Recover(err))
 			}
+			// Allow log/printf output to flush to journalctl, without this,
+			// it appears that some log output is lost.
+			time.Sleep(time.Second)
```

Comment:
Can we take this out? I don't see how this is an issue anymore: while we may be dropping log messages, the pertinent message should still make its way to the user since we're propagating the error properly now, without relying on seeing remote worker error messages along the way...

Reply:
ok, I took it out, I created an issue to track this. As an aside, I have another app here where we're not seeing an error remotely that I'm pretty sure occurs and that I can see locally.

Reply:
Strange.  I have a hard time seeing how a sleep on the worker would make a difference here; is something killing the process/worker immediately?  Otherwise, it seems like there would be plenty of time for any logs to flush.  Or maybe it's that the coordinator program quits too soon after the error, so the ssh tail does not receive it before quitting?

Reply:
I think it's the latter. My working theory is that the log messages are still winding their way through the local systemd/journalctl buffering and the journalctl -f that the master executes via ssh. The delay simply gives the the tail -f more time to see the data.
Unfortunately journalctl -f will never exit of it's own accord, even if the requested unit doesn't exist, because, you know, it may exist in the future! An explicit synchronisation method would be the only end the tail when systemctl status reports the process as non-existent. To make this work well, I think we'd want to not use journalctl directly but rather write out own wrapper that behaves as we want - i.e. tail -f whilst the worker is up, then exit cleanly. The wrapper would simple call journalctl and monitor the process and if it fails, exit etc. It could be a simple go app, but of course it would need to be copied over the worker. We should probably move this discussion to #23.